pydantic/main.py

```diff
@@ -675,9 +675,18 @@ def __get_validators__(cls) -> 'CallableGenerator':
     @classmethod
     def validate(cls: Type['Model'], value: Any) -> 'Model':
         if isinstance(value, cls):
-            if cls.__config__.copy_on_model_validation:
+            if isinstance(cls.__config__.copy_on_model_validation, bool):
+                # Warn about deprecated behavior
+                warnings.warn('`copy_on_model_validation` should be a string of value ("deep", "shallow", "none")', RuntimeWarning)
+
+            if cls.__config__.copy_on_model_validation == "shallow":
```

Comment:
This is introducing another breaking change. `copy_on_model_validation` being a bool should restore the pre-1.9.1 behavior. 

```suggestion
            if isinstance(cls.__config__.copy_on_model_validation, bool):
                # Warn about deprecated behavior
                warnings.warn('`copy_on_model_validation` should be a string of value ("deep", "shallow", "none")', RuntimeWarning)

            if cls.__config__.copy_on_model_validation == "shallow" or cls.__config__.copy_on_model_validation is True:
```

Reply:
@AngellusMortis I think I've resolved this.